File: data/src/main/java/com/somename/data/repository/datasource/mapper/ThumbnailEntityMapper.java
package com.somename.data.repository.datasource.mapper;

import com.somename.data.сontent.ThumbnailEntity;
import com.somename.domain.model.Thumbnail;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class ThumbnailEntityMapper extends Mapper<Thumbnail, ThumbnailEntity> {

    @Inject
    public ThumbnailEntityMapper() {
    }

    @Override
    public ThumbnailEntity map(Thumbnail value) {
        throw new UnsupportedOperationException();
    }

    @Override
    public Thumbnail reverseMap(ThumbnailEntity thumbnailEntity) {
        Thumbnail thumbnail = new Thumbnail();
        thumbnail.setId(thumbnailEntity.getId());
        thumbnail.setImageUrl(thumbnailEntity.getImageUrl());
        return thumbnail;
    }
}

File: data/src/main/java/com/somename/data/repository/datasource/mapper/ThumbnailEntityMapper.kt
package com.somename.data.repository.datasource.mapper


import com.somename.data.сontent.ThumbnailEntity
import com.somename.domain.model.Thumbnail
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class ThumbnailEntityMapper @Inject
constructor() : Mapper<Thumbnail, ThumbnailEntity>() {

    override fun map(value: Thumbnail): ThumbnailEntity {
        throw UnsupportedOperationException()
    }

    override fun reverseMap(value: ThumbnailEntity): Thumbnail {
        val thumbnail = Thumbnail()
        thumbnail.id = value.id
        thumbnail.imageUrl = value.imageUrl
        return thumbnail
    }
}
